### PR TITLE
MWPW-135656: refactor region selector and social links decoration

### PR DESCRIPTION
**Pull request by robert-bogos**
## Description
This helps refactor the region selector and social links decoration in `global-footer.js`

## Related Issue
Resolves: [MWPW-135656](https://jira.corp.adobe.com/browse/MWPW-135656)

## Test URLs
**Acrobat:**
- Before: https://main--dc--adobecom.hlx.page/drafts/rbogos/default?martech=off
- After: https://main--dc--adobecom.hlx.page/drafts/rbogos/default?milolibs=global-footer-decoration--milo--robert-bogos&martech=off

**BACOM:**
- Before: https://main--bacom--adobecom.hlx.page/drafts/rbogos/default
- After: https://main--bacom--adobecom.hlx.page/drafts/rbogos/default?milolibs=global-footer-decoration--milo--robert-bogos&martech=off

**CC:**
- Before: https://main--cc--adobecom.hlx.page/drafts/rbogos/default?martech=off
- After: https://main--cc--adobecom.hlx.page/drafts/rbogos/default?milolibs=global-footer-decoration--milo--robert-bogos&martech=off

**Milo:**
- Before: https://main--milo--robert-bogos.hlx.page/drafts/rbogos/default?martech=off
- After: https://global-footer-decoration--milo--robert-bogos.hlx.page/drafts/rbogos/default?martech=off

### CODE REVIEW

File: libs/blocks/global-footer/global-footer.js [hunk 1]
```diff
@@ -69,16 +69,38 @@ class Footer {
...
-    // TODO: revisit region picker and social links decoration logic
+
     const regionAnchor = this.body.querySelector('.region-selector a');
-    if (regionAnchor?.href) {
-      regionAnchor.setAttribute('href', `${regionAnchor.getAttribute('href')}#_dnt#_dnb`);
+    const regionNextSibling = regionAnchor?.nextSibling;
+    const regionParent = regionAnchor?.parentNode;
+    const detachedRegion = regionParent?.removeChild(regionAnchor);
+
+    const socialLinks = Array.from(document.querySelectorAll('.social a'));
```
Comment (narcis-radu) [inline libs/blocks/global-footer/global-footer.js:78]:
```suggestion
    const socialLinks = [...document.querySelectorAll('.social a')];
```

Comment (narcis-radu) [reply libs/blocks/global-footer/global-footer.js:78]: `socialLinks` returns as an empty array for me

File: libs/blocks/global-footer/global-footer.js [hunk 2]
```diff
@@ -69,16 +69,38 @@ class Footer {
 
     // TODO: log to LANA if Footer content could not be found
     if (!this.body) return;
-    // TODO: revisit region picker and social links decoration logic
+
     const regionAnchor = this.body.querySelector('.region-selector a');
-    if (regionAnchor?.href) {
-      regionAnchor.setAttribute('href', `${regionAnchor.getAttribute('href')}#_dnt#_dnb`);
+    const regionNextSibling = regionAnchor?.nextSibling;
```
Comment (narcis-radu) [inline libs/blocks/global-footer/global-footer.js:74]: This element seems to be empty all the time, maybe we can remove it if not needed.

File: libs/blocks/global-footer/global-footer.js (hunk 1 above)
Comment (robert-bogos) [reply libs/blocks/global-footer/global-footer.js:78]: Fixed

File: libs/blocks/global-footer/global-footer.js (hunk 2 above)
Comment (robert-bogos) [reply libs/blocks/global-footer/global-footer.js:74]: It was just an extra precaution, but not actually needed. I removed it

File: libs/blocks/global-footer/global-footer.js [hunk 3]
```diff
@@ -69,17 +69,25 @@ class Footer {
...
+    );
+
     decorateLinks(this.body);
 
+    if (detachedRegion && regionParent) regionParent.appendChild(detachedRegion);
+
+    detachedSocialLinks.forEach((link, index) => {
+      if (link && socialParents[index]) socialParents[index].appendChild(link);
+    });
+
```
Comment (overmyheadandbody) [inline libs/blocks/global-footer/global-footer.js:90]: Detaching the links momentarily could be a solution to this, but I feel this is a bit too verbose. Could we not simplify with something like:
```
const [ region, social ] = ['.region-selector', '.social'].map((selector) => this.body.querySelector(selector));
const [ regionParent, socialParent ] = [ region.parentElement, social.parentElement ];
[ regionParent, socialParent ].forEach((parent) => parent.replaceChildren());

decorateLinks(this.body);

regionParent.appendChild(region);
socialParent.appendChild(social);
```

Comment (robert-bogos) [reply libs/blocks/global-footer/global-footer.js]: Done. Indeed is simpler like this

**Review by overmyheadandbody — APPROVED** [review]

**Comment by overmyheadandbody** [comment]
We should also look into whether we have unit tests to ensure correct behaviour in these sections wrt skipping initial link decoration.

**Review by narcis-radu — APPROVED** [review]

**Review by SilviuLCF — APPROVED** [review]

**Pull request merged**